### PR TITLE
Move store & keyStore

### PR DESCRIPTION
**Pull request by basine**
We need to move the database and the keystore in a different directory that is unique for the account. We use the a UUID (e.g. the user remoteIdentifier) to distinguish and access stores of different accounts.

NOTE: This is an API breaker

### CODE REVIEW

File: Source/ManagedObjectContext/PersistentStoreRelocator.swift [hunk 1]
```diff
@@ -18,28 +18,69 @@
...
+    @objc(URLByAppendingStorePath)
+    public func appendingStorePath() -> NSURL {
+        return (self as URL).appendingStorePath() as NSURL
+    }
+}
+
+
+extension URL {
+    /// Appends the path to the persistentStore in the form baseURL/{bundleId}/{accountIdentifier}/store.wiredatabase
+    public func appendingStorePath() -> URL {
```
Comment (daehn) [inline Source/ManagedObjectContext/PersistentStoreRelocator.swift]: In the comment it says that the form will be `baseURL/{bundleId}/{accountIdentifier}/store.wiredatabase` but it looks like it is `baseURL/{bundleId}/store.wiredatabase`.

File: Tests/Source/Model/Utils/FileManager+FileLocationTests.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            let lastComp = components.removeLast()
+            XCTAssertEqual(lastComp, "Caches")
+        } else {
+            XCTFail("WARNING!! Changing the method `appendingStorePath` might break migration. See PersistentStoreRelocator")
+        }
+    }
+}
+
+
+class FileManager_FileLocationTests : XCTestCase {
```
Comment (vytis) [inline Tests/Source/Model/Utils/FileManager+FileLocationTests.swift]: Probably can be deleted

**Comment by vytis** [comment]
Also according to our brand-new test coverage tracking tool [Source/Utilis/FileManager+FileLocations.swift](https://codecov.io/gh/wireapp/wire-ios-data-model/pull/317?src=pr&el=tree#diff-U291cmNlL1V0aWxpcy9GaWxlTWFuYWdlcitGaWxlTG9jYXRpb25zLnN3aWZ0) could use some more tests, only 31% of the file is covered.

**Comment by typfel** [comment]
Would it be possible to bake the `accountId` into the `URL`which passed in? Or is the sharedContainer URL needed for migration?

**Comment by basine** [comment]
@typfel the sharedContainer is necessary for the migration

**Comment by mikeger** [comment]
@daehn @basine I thought we wanted to merge this one yesterday, any obstacles / TODOs left?

**Comment by daehn** [comment]
@mikeger @typfel @vytis We just merged the rebased changes from the stack initialisation branch into this one, which now uses the `StorageStack` internally.

File: Tests/Source/Model/ZMBaseManagedObjectTest.m [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 #import "NSString+RandomString.h"
 
-NSString *const ZMPersistedClientIdKey = @"PersistedClientId";
+//NSString *const ZMPersistedClientIdKey = @"PersistedClientId";
```
Comment (mikeger) [inline Tests/Source/Model/ZMBaseManagedObjectTest.m]: Not necessary any more / can be removed?

File: Tests/Source/Model/ZMTestSession.h [hunk 4]
```diff
@@ -62,7 +63,7 @@
 - (void)performPretendingUiMocIsSyncMoc:(void(^)(void))block;
 
 /// Wait until all contexts have completed their task and tear them down
-- (void)waitAndDeleteAllManagedObjectContexts;
+//- (void)waitAndDeleteAllManagedObjectContexts;
```
Comment (mikeger) [inline Tests/Source/Model/ZMTestSession.h]: Can be removed?

**Pull request merged**